src/ott/problems/nn/dataset.py

```diff
@@ -0,0 +1,155 @@
+
+  Args:
+  trainloader_source: Training dataset, source measure
+  trainloader_target: Training dataset, target measure
+  validloader_source: Valid dataset, source measure
+  validloader_target: Valid dataset, target measure
+  """
+  trainloader_source: Iterable[jnp.ndarray]
+  trainloader_target: Iterable[jnp.ndarray]
+  validloader_source: Iterable[jnp.ndarray]
+  validloader_target: Iterable[jnp.ndarray]
+
+
+GAUSSIAN_CENTERS = {
+    'simple':
+        np.array([[0, 0]]),
+    'circle':
+        np.array([
+            (1, 0),
+            (-1, 0),
+            (0, 1),
+            (0, -1),
+            (1.0 / np.sqrt(2), 1.0 / np.sqrt(2)),
+            (1.0 / np.sqrt(2), -1.0 / np.sqrt(2)),
+            (-1.0 / np.sqrt(2), 1.0 / np.sqrt(2)),
+            (-1.0 / np.sqrt(2), -1.0 / np.sqrt(2)),
+        ]),
+    'square_five':
+        np.array([[0, 0], [1, 1], [-1, 1], [-1, -1], [1, -1]]),
+    'square_four':
+        np.array([[1, 0], [0, 1], [-1, 0], [0, -1]]),
+}
+
+
+@dataclasses.dataclass
+class GaussianMixture:
+  """A mixture of Gaussians.
+
+  Args:
+    name: the name specifying the centers of the mixture components;
+      `simple` (data clustered in one center),
+      `circle` (two-dimensional Gaussians arranged on a circle),
+      `square_five` (two-dimensional Gaussians on a square with one Gaussian
+      in the center), and `square_four` (two-dimensional Gaussians in the
+      corners of a rectangle)
+    batch_size: batch size of the samples
+    init_key: initial PRNG key
+    scale: scale of the individual Gaussian samples
+    variance the variance of the individual Gaussian samples
+  """
+  name: str
+  batch_size: int
+  init_key: jax.random.PRNGKey
+  scale: float = 5.0
+  variance: float = 0.5
+
+  def __post_init__(self):
+    if self.name not in GAUSSIAN_CENTERS:
+      raise ValueError(
+          f'{self.name} is not a valid dataset for GaussianMixture'
+      )
+    self.centers = GAUSSIAN_CENTERS[self.name]
+
+  def __iter__(self):
+    return self.create_sample_generators()
+
+  def create_sample_generators(self) -> Iterator[jnp.array]:
+    # create generator which randomly picks center and adds noise
+    key = self.init_key
+    while True:
+      k1, k2, key = jax.random.split(key, 3)
+      means = jax.random.choice(k1, self.centers, [self.batch_size])
+      normal_samples = jax.random.normal(k2, [self.batch_size, 2])
+      samples = self.scale * means + self.variance ** 2 * normal_samples
+      yield samples
+
+
+def get_sampler(name: str, **kwargs: Any) -> Iterable[jnp.ndarray]:
+  """Returns a sampler.
+
+  Args:
+    name: the name of the dataset
+    kwargs: additional arguments passed into the sampler
+
+  Returns:
+    The sampler
+  """
+  if name.startswith('gaussian'):
+    name = name[9:]
+    return iter(GaussianMixture(name, **kwargs))
+  else:
+    raise ValueError('Only Gaussian datasets are supported')
+
+
+def create(
```

Comment:
I'd rename this (or explain in docstring) this is toy/example data.

Reply:
I renamed it to `create_gaussian_mixture_samplers` (and if we add more data we can make something a little more general)